glslc/test/placeholder.py

```diff
@@ -115,3 +116,19 @@ def instantiate_for_glslc_args(self, testcase):
 
     def instantiate_for_expectation(self, testcase):
         return os.path.join(testcase.directory, self.filename)
+
+
+class SpecializedString(PlaceHolder):
+    """Returns a string that has been specialized based on TestCase."""
+
+    def __init__(self, filename):
```

Comment:
Unfortunately:
glslc/test/glslc_test_framework.py
requires that this parameter is "filename".

I plan on adding a follow-up CL in the future that changes this.